src/function_classes/linear.py

```diff
@@ -33,7 +33,7 @@ def __init__(self, sparsity: int = 3, *args, **kwargs):
         self._sparsity = sparsity
 
     def evaluate(self, x_batch: torch.Tensor, *params: torch.Tensor) -> torch.Tensor:
-        weights, _ = params
+        weights, *_ = params
```

Comment:
the correct fix here is `weights = torch.stack(params)` I believe, but this is fixed anyway in #31

Reply:
I think this is right, review that PR